### PR TITLE
Handling special (regex) characters and multiple test names in specs2

### LINKED ISSUE #355: Support test names with special characters and braces in specs2
The specs2 test runner does not currently handle unescaped regex input (which may be invalid due to any unsafe characters may appear as part of the test name).

This is already handled by the Bazel plugin - the test filter is escaped (using `Pattern.quote`) before sent to bazel.

### PR DESCRIPTION
**Pull request by hmemcpy**
This fixes #355

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by hmemcpy** [comment]
@ittaiz please review

**Comment by ittaiz** [comment]
Test this please

File: src/java/io/bazel/rulesscala/specs2/Specs2RunnerBuilder.scala [hunk 1]
```diff
@@ -66,15 +87,15 @@ class Specs2ClassRunner(testClass: Class[_], testFilter: Pattern)
 
   private def specs2ExamplesMatching(testFilter: Pattern, junitDescription: Description): Option[String] =
     flattenDescription(junitDescription)
-      .filter(matching(testFilter))
```
Comment (ittaiz) [inline src/java/io/bazel/rulesscala/specs2/Specs2RunnerBuilder.scala]: why the change from filter to find?

Comment (hmemcpy) [reply src/java/io/bazel/rulesscala/specs2/Specs2RunnerBuilder.scala]: Hmm, a leftover from trying different things. Will fix, thanks!

Comment (ittaiz) [reply src/java/io/bazel/rulesscala/specs2/Specs2RunnerBuilder.scala]: Maybe we need another test?

Comment (hmemcpy) [reply src/java/io/bazel/rulesscala/specs2/Specs2RunnerBuilder.scala]: Perhaps. The test filter string doesn't currently support passing more than one test method, so currently there isn't a difference between `find` and `filter`. To be honest, I don't even know how to specify multiple test names in intellij...

**Comment by ittaiz** [comment]
can't a regex utilize this? to match multiple ones?
Also this interface is for intellij and command line

**Comment by hmemcpy** [comment]
Yeah, the regex should be able to handle multiple test methods, e.g. `foo should::(do bar|do baz)`, but with the escaping, they have to be individually escaped, i.e.: `foo should::(\Qdo bar\E|\Qdo baz\E)`.

Most of this is already supported (the intellij plugin now escapes individual test names). It needs to be tested and tweaked, if necessary, but perhaps this should be an explicit issue.

**Comment by ittaiz** [comment]
Ok, if so then please leave it as find and open a detailed issue about this
missing feature so people will know it's only partly supported

**Comment by ittaiz** [comment]
test this please

**Comment by ittaiz** [comment]
test this please
